glide-core/tests/test_socket_listener.rs

```diff
@@ -619,6 +619,83 @@ mod socket_listener {
+        let mut request = get_command_request(
+            CALLBACK_INDEX,
+            vec!["INFO".to_string(), "SERVER".to_string()],
+            RequestType::CustomCommand.into(),
+            false,
+        );
+        let mut routes = redis_request::Routes::default();
+        routes.set_simple_routes(redis_request::SimpleRoutes::Random);
+        request.route = Some(routes).into();
+        write_message(&mut buffer, request);
+        test_basics.socket.write_all(&buffer).unwrap();
+
+        let _size = read_from_socket(&mut buffer, &mut test_basics.socket);
+        let (message_length, header_bytes) = parse_header(&buffer);
+        let response = decode_response(&buffer, header_bytes, message_length as usize);
+
+        assert_eq!(response.callback_idx, CALLBACK_INDEX);
+        let Some(response::Value::RespPointer(pointer)) = response.value else {
+            panic!("Unexpected response {:?}", response.value);
+        };
+        let pointer = pointer as *mut Value;
+        let received_value = unsafe { Box::from_raw(pointer) };
+        let first_value = String::from_redis_value(&received_value).unwrap();
+        let port = first_value
+            .split("\r\n")
+            .filter_map(|str| {
+                if str.starts_with("tcp_port:") {
+                    let split = str.split(':').nth(1).unwrap();
+                    split.parse::<u16>().ok()
+                } else {
+                    None
+                }
+            })
+            .next()
+            .unwrap();
+
+        buffer.clear();
+        let mut request = get_command_request(
+            CALLBACK_INDEX,
+            vec!["INFO".to_string(), "SERVER".to_string()],
+            RequestType::CustomCommand.into(),
+            false,
+        );
+        let mut routes = redis_request::Routes::default();
+        let by_address_route = glide_core::redis_request::ByAddressRoute {
+            host: "127.0.0.1".into(),
+            port: port as i32,
+            ..Default::default()
+        };
+        routes.set_by_address_route(by_address_route);
+        request.route = Some(routes).into();
+        write_message(&mut buffer, request);
+        test_basics.socket.write_all(&buffer).unwrap();
+
+        let _size = read_from_socket(&mut buffer, &mut test_basics.socket);
+        let (message_length, header_bytes) = parse_header(&buffer);
+        let response = decode_response(&buffer, header_bytes, message_length as usize);
+
+        assert_eq!(response.callback_idx, CALLBACK_INDEX);
+        let Some(response::Value::RespPointer(pointer)) = response.value else {
+            panic!("Unexpected response {:?}", response.value);
+        };
+        let pointer = pointer as *mut Value;
+        let received_value = unsafe { Box::from_raw(pointer) };
+        let second_value = String::from_redis_value(&received_value).unwrap();
+        // 400 characters, because after that appears time-based values.
```

Comment:
It can be easily break by any new info entries.. the best is to run cluster nodes and to make sure that the 'myself' flag is corresponding to the correct address